# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/generic/DocAccessible.cpp
# Commit: c7abc15f7444
# Full Hash: c7abc15f7444c9ebc5c6adafc3a1be2d5d1e3090
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2019-09-07 09:40:21
# Description:
#   Bug 1578282 - Use flattened tree parent when getting inserted node container. r=Jamie
#   
#   This assures that a node that has already been inserted in a shadow root
#   subtree does not get erroneously reinserted into the top container.
#   
# ==============================================================================

diff -r 96405b0b1298 -r c7abc15f7444 accessible/generic/DocAccessible.cpp
--- a/accessible/generic/DocAccessible.cpp	Fri Sep 06 17:35:48 2019 +0000
+++ b/accessible/generic/DocAccessible.cpp	Fri Sep 06 00:24:21 2019 +0000
@@ -1830,8 +1830,8 @@
       continue;
     }
 
-    Accessible* container =
-        Document()->AccessibleOrTrueContainer(node->GetParentNode(), true);
+    Accessible* container = Document()->AccessibleOrTrueContainer(
+        node->GetFlattenedTreeParentNode(), true);
     if (container != Context()) {
       continue;
     }